### PR TITLE
Report test framework data from child processes instead of parsing project dependencies

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
# What Does This Do
Update CI Visibility: test framework name and version are now sent from children processes to the parent process using signal server.

# Motivation
Improving the accuracy of test framework name and version detection.

Test session and test modules spans are created by the parent process.
These spans need to be tagged with test framework name and version.
Previous implementation attempted to extract that data by parsing the list of project's Maven or Gradle dependencies.
That approach often gave inaccurate results (for instance, in cases when a project had JUnit dependency but ran another framework such as Spock or Cucumber on top of JUnit).
The new approach is more accurate since the data is obtained from test framework instrumentations.

### CODE REVIEW

File: dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/context/ParentProcessTestContext.java [hunk 1]
```diff
@@ -30,4 +33,34 @@ public Long getParentId() {
   public AgentSpan getSpan() {
     return null;
   }
+
+  @Override
+  public void reportChildTag(String key, Object value) {
+    // the method leaves room for a
+    // proper implementation using a thread-safe map,
+    // but for not it's just this,
+    // in order to save some performance costs
```
Comment (juan-fernandez) [inline dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/context/ParentProcessTestContext.java:42]: nitpick 😄 
```suggestion
    // but for now it's just this,
    // to save some performance costs
```

File: dd-java-agent/agent-ci-visibility/src/test/groovy/datadog/trace/civisibility/ipc/ModuleExecutionResultTest.groovy [hunk 2]
```diff
@@ -16,11 +16,11 @@ class ModuleExecutionResultTest extends Specification {
...
-      new ModuleExecutionResult(12345, 67890, false, false, 0),
-      new ModuleExecutionResult(12345, 67890, true, false, 1),
-      new ModuleExecutionResult(12345, 67890, false, true, 2),
-      new ModuleExecutionResult(12345, 67890, false, false, 3),
-      new ModuleExecutionResult(12345, 67890, true, true, Integer.MAX_VALUE)
+      new ModuleExecutionResult(12345, 67890, false, false, 0, null, null),
+      new ModuleExecutionResult(12345, 67890, true, false, 1, "abc", "def"),
+      new ModuleExecutionResult(12345, 67890, false, true, 2, null, "def"),
+      new ModuleExecutionResult(12345, 67890, false, false, 3, "abc", null),
+      new ModuleExecutionResult(12345, 67890, true, true, Integer.MAX_VALUE, "abc", "def")
```
Comment (juan-fernandez) [inline dd-java-agent/agent-ci-visibility/src/test/groovy/datadog/trace/civisibility/ipc/ModuleExecutionResultTest.groovy:23]: do the last two values represent `testFramwork` and `testFrameworkVersion`? If that's the case, is there value in setting more real values? Just to make it easier to read

Comment (juan-fernandez) [reply dd-java-agent/agent-ci-visibility/src/test/groovy/datadog/trace/civisibility/ipc/ModuleExecutionResultTest.groovy:23]: if doing this implies doing a bigger refactor feel free to ignore it

File: dd-java-agent/instrumentation/maven-3.2.1/src/test/groovy/MavenTest.groovy [hunk 3]
```diff
@@ -279,12 +279,12 @@ class MavenTest extends CiVisibilityTest {
...
   @Override
   String expectedTestFramework() {
-    return "junit4"
+    return null
   }
 
   @Override
   String expectedTestFrameworkVersion() {
-    return "4.13.2"
+    return null
```
Comment (juan-fernandez) [inline dd-java-agent/instrumentation/maven-3.2.1/src/test/groovy/MavenTest.groovy:287]: why is this now null?

**Review by juan-fernandez — APPROVED** [review]

File: dd-java-agent/agent-ci-visibility/src/test/groovy/datadog/trace/civisibility/ipc/ModuleExecutionResultTest.groovy (hunk 2 above)
Comment (nikita-tkachenko-datadog) [reply dd-java-agent/agent-ci-visibility/src/test/groovy/datadog/trace/civisibility/ipc/ModuleExecutionResultTest.groovy:23]: I don't think there's much value other than some readability gains. Still, updated this

File: dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/context/ParentProcessTestContext.java (hunk 1 above)
Comment (nikita-tkachenko-datadog) [reply dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/context/ParentProcessTestContext.java:42]: Updated

**Pull request merged**